api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -214,18 +220,23 @@ public async Task<IActionResult> UpdateAsync(ReportModel model, [FromQuery] bool
         if (result?.OwnerId != user?.Id) throw new NotAuthorizedException("Not authorized to update this report");
         _reportService.ClearChangeTracker(); // Remove the report from context.
         result = _reportService.Update(model.ToEntity(_serializerOptions));
-        if (updateInstances && model.Instances.Any())
+        var instanceModel = model.Instances.FirstOrDefault();
+        Entities.ReportInstance? instance = null;
+        if (updateInstances && instanceModel != null)
         {
-            // There should only ever be one instance returned for the current user.
-            model.Instances.ForEach(instance =>
-            {
-                var updatedInstance = _reportInstanceService.Update((Entities.ReportInstance)instance);
-            });
+            // Only update the first instance.
+            instance = _reportInstanceService.Update((Entities.ReportInstance)instanceModel);
         }
         _reportInstanceService.CommitTransaction();
         var report = _reportService.FindById(result.Id) ?? throw new NoContentException("Report does not exist");
+        if (updateInstances && instance != null)
+        {
+            report.Instances.Add(instance);
+            instance.ContentManyToMany.Clear();
+            instance.ContentManyToMany.AddRange(_reportInstanceService.GetContentForInstance(instance.Id));
```

Comment:
To improve performance content is not automatically returned by a report instance, it needs to be fetched separately.